### PR TITLE
Feat tabs design changes

### PR DESCRIPTION
**Pull request by bruno-sch**
## Proposed changes

Implementation of the design changes to the tabs. As part of this, a small refactoring of the scss code.

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: __snapshots__/tabs/showcase/mobile-chrome/regular/neutral/DBTabs-should-match-screenshot.png
Comment (nmerget) [inline __snapshots__/tabs/showcase/mobile-chrome/regular/neutral/DBTabs-should-match-screenshot.png:1]: There is some gray background for the tab items, this seems wrong

File: packages/components/src/components/tab/tab.scss [hunk 1]
```diff
@@ -3,63 +3,77 @@
...
+			&::before {
+				@extend %focus-placeholder;
+
+				content: "";
+				position: absolute;
+				inset-block-start: 0;
+				inset-inline-start: 0;
+				inline-size: 100%;
+				block-size: 100%;
+				outline-offset: variables.$db-spacing-fixed-2xs;
```
Comment (nmerget) [inline packages/components/src/components/tab/tab.scss:60]: Why did you change the offset? This would lead to an inconsistent behaviour between the remaining components, right?

File: packages/components/src/components/tab/tab.scss [hunk 2]
```diff
@@ -3,63 +3,77 @@
...
+		/*
+		focus must be set on input:
+		- parent label tag may use before AND after for displaying icons
+		- placed on li.db-tab focus would get cut off partially by by overflow hidden (orientation vertical)
+		 */
+		&:focus-visible {
+			&::before {
+				@extend %focus-placeholder;
+
+				content: "";
```
Comment (nmerget) [inline packages/components/src/components/tab/tab.scss:54]: If I remove this `::before` the default focus for the `input` works out-of-the-box? Do we need this here?

File: showcases/react-showcase/src/components/form/index.tsx [hunk 3]
```diff
@@ -240,6 +240,17 @@ const FormComponent = () => {
 					))}
 				</DBAccordion>
 
+				<br />
+				<br />
+				<br />
```
Comment (mfranzke) [inline showcases/react-showcase/src/components/form/index.tsx:245]:
```suggestion
				<br />
```

File: showcases/react-showcase/src/components/form/index.tsx [hunk 4]
```diff
@@ -240,6 +240,17 @@ const FormComponent = () => {
...
+
+				<DBTab>Test 1</DBTab>
+
+				<br />
+				<br />
+				<br />
+				<br />
```
Comment (mfranzke) [inline showcases/react-showcase/src/components/form/index.tsx:252]:
```suggestion
				<br />
```

File: showcases/react-showcase/src/components/form/index.tsx [hunk 5]
```diff
@@ -251,6 +262,64 @@ const FormComponent = () => {
 					<DBTabPanel>Tab Panel 3</DBTabPanel>
 				</DBTabs>
 
+				<br />
+				<br />
+				<br />
```
Comment (mfranzke) [inline showcases/react-showcase/src/components/form/index.tsx:267]:
```suggestion
				<br />
```

File: showcases/react-showcase/src/components/form/index.tsx [hunk 6]
```diff
@@ -251,6 +262,64 @@ const FormComponent = () => {
...
+						right to find fault with a man who chooses to enjoy a
+						pleasure that has no annoying consequences, or one who
+						avoids a pain that produces no resultant pleasure?
+					</DBTabPanel>
+					<DBTabPanel>Tab Panel 5</DBTabPanel>
+				</DBTabs>
+
+				<br />
+				<br />
+				<br />
```
Comment (mfranzke) [inline showcases/react-showcase/src/components/form/index.tsx:321]:
```suggestion
				<br />
```

**Review by nmerget — APPROVED** [review]
